charts/jobs/templates/_helpers.tpl

```diff
@@ -79,10 +79,10 @@ Create the name of the service account to use
 {{- $podSecurityContext := $v.podSecurityContext | default (dict "runAsNonRoot" true "runAsUser" 1001 "runAsGroup" 1001) -}}
 {{- $containers := list (dict "isInit" false "container" $v) }}
 {{- $hasMounts := or $v.files $v.secretMounts }}
-{{- if $v.init }}
-  {{- $containers = prepend $containers (dict "isInit" true "container" $v.init) }}
-  {{- if or $v.init.files $v.init.SecretMounts }}{{ $hasMounts = true }}{{ end }}
-{{ end }}
+{{- range $vi := $v.init }}
+  {{- $containers = prepend $containers (dict "isInit" true "container" $vi) }}
```

Comment:
having two separate lists `containers` and `initContainers` would be a much simpler solution.
Is there any reason that you decided to not doing it?

Reply:
I understand your comment, I just used what was there. I can refactor it if you want.